src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -0,0 +1,514 @@
+      new HashSet<>(Arrays.asList(ColumnPhysicalType.SB8, ColumnPhysicalType.SB16));
+
+  private final Map<String, Pair<ColumnMetadata, Integer>> fieldIndex;
+  private final Map<String, String> metadata;
+  private final List<List<Object>> data;
+  private final List<List<Object>> tempData;
+
+  private MessageType schema;
+
+  ParquetRowBuffer(
+      SnowflakeStreamingIngestChannelInternal<ParquetRowBuffer.ParquetChunkData> channel) {
+    super(channel);
+    fieldIndex = new HashMap<>();
+    metadata = new HashMap<>();
+    data = new ArrayList<>();
+    tempData = new ArrayList<>();
+  }
+
+  @Override
+  public void setupSchema(List<ColumnMetadata> columns) {
+    fieldIndex.clear();
+    metadata.clear();
+    List<Type> parquetTypes = new ArrayList<>();
+    int id = 1;
+    for (ColumnMetadata column : columns) {
+      Type type = buildField(column, id);
+      parquetTypes.add(type);
+      fieldIndex.put(column.getName(), new Pair<>(column, parquetTypes.size() - 1));
+      if (!column.getNullable()) {
+        this.nonNullableFieldNames.add(column.getName());
+      }
+      this.statsMap.put(column.getName(), new RowBufferStats(column.getCollation()));
+
+      if (this.owningChannel.getOnErrorOption() == OpenChannelRequest.OnErrorOption.ABORT) {
+        this.tempStatsMap.put(column.getName(), new RowBufferStats(column.getCollation()));
+      }
+
+      id++;
+    }
+    schema = new MessageType("bdec", parquetTypes);
+  }
+
+  private Type buildField(ColumnMetadata column, int id) {
+    Type parquetType;
+    String name = column.getName();
+
+    ColumnPhysicalType physicalType;
+    ColumnLogicalType logicalType;
+    try {
+      physicalType = ColumnPhysicalType.valueOf(column.getPhysicalType());
+      logicalType = ColumnLogicalType.valueOf(column.getLogicalType());
+    } catch (IllegalArgumentException e) {
+      throw new SFException(
+          ErrorCode.UNKNOWN_DATA_TYPE, column.getLogicalType(), column.getPhysicalType());
+    }
+
+    this.metadata.put(
+        Integer.toString(id), logicalType.getOrdinal() + "," + physicalType.getOrdinal());
+
+    LogicalTypeAnnotation.DecimalLogicalTypeAnnotation decimalLogicalTypeAnnotation =
+        column.getScale() != null && column.getPrecision() != null
+            ? LogicalTypeAnnotation.DecimalLogicalTypeAnnotation.decimalType(
+                column.getScale(), column.getPrecision())
+            : null;
+
+    // Handle differently depends on the column logical and physical types
+    Type.Repetition repetition =
+        column.getNullable() ? Type.Repetition.OPTIONAL : Type.Repetition.REQUIRED;
+    switch (logicalType) {
+      case FIXED:
+        parquetType =
+            getFixedParquetType(
+                column, id, name, physicalType, decimalLogicalTypeAnnotation, repetition);
+        break;
+      case ARRAY:
+      case OBJECT:
+      case VARIANT:
+        this.metadata.put(id + ":obj_enc", "1");
```

Comment:
no `parquertType` for this case?

Reply:
parquetType is same as the next one for ANY. CHAR etc
the object types needs this flag for the scanner